auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -512,6 +560,18 @@ interface CredentialParseFunction {
         void parseCredential(String algorithm, String format, String body) throws RealmUnavailableException, XMLStreamException;
     }
 
+    // only to be used by wildfly when :write-attribute for key-store, or key-store alias is called
+    public void updateRealmKeyPair() throws IOException, GeneralSecurityException {
+        ModifiableRealmIdentityIterator realmIterator = this.getRealmIdentityIterator();
+        while (realmIterator.hasNext()) {
+            Identity identity = (Identity) realmIterator.next();
+            identity.createDigitalSignature(String.valueOf(identity.path), identity.name);
+            Identity.updateMainIndex(identity.mainIndex, identity.path, identity.secretKey);
```

Comment:
Currently, this block is iterating over each identity, and calling `updateMainIndex` each time, which in turn reads the index file, searches for the entry that needs to be updated, and updates it. All of the entries need to be updated because the key pair has changed, right? It would be better to just write the new entries all at once.

Reply:
That makes sense, will start the main index from scratch instead